server/test/src/scala/lsug/markup/ParserSpec.scala

```diff
@@ -186,4 +192,37 @@ trait ParserChecks extends LsugSuite {
     checkFailure(product(left, right), text)
       .label("product")
       .build(name)
+
+  implicit def arbResult[T](implicit a: Arbitrary[T]): Arbitrary[Result[T]] =
+    Arbitrary(
+      Gen.oneOf(
+        Gen.const(Result.Fail),
+        for {
+          e <- Arbitrary.arbitrary[T]
+          s <- Arbitrary.arbitrary[String]
+        } yield Result.Success[T](e, s)
+      )
+    )
+
+  implicit def parserEq[A: Eq]: Eq[Parser[A]] =
+    Eq.fromUniversalEquals[Parser[A]]
+
+  implicit lazy val arbRegex: Arbitrary[Regex] = Arbitrary {
+    val lsugRegex: List[Regex] =
+      List("◊".r, "\\{".r, "\\}".r, "[a-z]+".r, "[^◊}]+".r)
+    Gen.oneOf(lsugRegex)
+  }
+
+  implicit def arbParser[T](implicit a: Arbitrary[T]): Arbitrary[Parser[T]] =
```

Comment:
Thanks @kxa14! 

 ```
[error] /home/zainab/scala/lsug-website/server/test/src/scala/lsug/markup/ParserSpec.scala:224:13: type mismatch;
[error]  found   : org.scalacheck.Gen[lsug.markup.Parser.Pattern]
[error]  required: org.scalacheck.Gen[lsug.markup.Parser[T]]
[error]           r <- Arbitrary.arbitrary[Regex]
 ```
 
 This is a bit cryptic, but it's a very real error. The `arbParser` function should construct a `Arbitrary[Parser[T]]` when given an `Arbitrary[T]`. For example, if you passed in an arbitrary number generator (`Arbitrary[Int]`) it should construct an arbitrary parser that generates integers on parse (`Arbitrary[Parser[Int]]`).
 
 However, the following code generates a `Parser[String]` no matter what the type `T` is.

 ```scala
 for {
   r <- Arbitrary.arbitrary[Regex]
 } yield Parser.Pattern(r)
 ```
 
 You couldn't use that parser to produce integers.
 
 One option would be to map over the `Parser[String]` to produce a `Parser[Int]`:

 ```scala
for {
  e <- Arbitrary.arbitrary[T]
  r <- Arbitrary.arbitrary[Regex]
} yield Parser.map(Parser.Pattern(r))(_ => e)
 ```
 
 This produces a valid `Arbitrary[Parser[T]]`